### PR TITLE
Switch context to env command

### PR DESCRIPTION
**Pull request by kaduartur**
### Description

All context commands have been deprecated and new envs created:

- `
rit set context` ➡️  `rit set env`
- `rit show context` ➡️  `rit show env`
- `rit delete context` ➡️  `rit delete env`

### How to verify it

All new commands must behave as the old commands worked, only with the difference that now instead of context we have environments.

### Changelog
Switch context to env command

### CODE REVIEW

File: internal/mocks/mocks.go [hunk 1]
```diff
@@ -17,19 +17,20 @@
...
 import (
-	"github.com/ZupIT/ritchie-cli/pkg/formula"
-	"github.com/ZupIT/ritchie-cli/pkg/rcontext"
 	"github.com/stretchr/testify/mock"
+
+	"github.com/ZupIT/ritchie-cli/pkg/env"
+	"github.com/ZupIT/ritchie-cli/pkg/formula"
 )
 
 type ContextFinderMock struct {
```
Comment (brunasilvazup) [inline internal/mocks/mocks.go:26]: This function should be `EnvFinderMock` right?

File: pkg/cmd/delete_credential.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func NewDeleteCredentialCmd(
 
 func (d deleteCredentialCmd) runPrompt() CommandRunnerFunc {
 	return func(cmd *cobra.Command, args []string) error {
-		context, err := d.getCurrentContext()
+		context, err := d.currentEnv()
```
Comment (brunasilvazup) [inline pkg/cmd/delete_credential.go:59]: This var `context` should be `env` right?

File: pkg/cmd/delete_credential.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func (d deleteCredentialCmd) runStdin() CommandRunnerFunc {
 			return err
 		}
 
-		context, err := d.getCurrentContext()
+		context, err := d.currentEnv()
```
Comment (brunasilvazup) [inline pkg/cmd/delete_credential.go:107]: Same here

File: pkg/cmd/delete_credential_test.go [hunk 4]
```diff
@@ -72,9 +72,9 @@ func TestDeleteCredential(t *testing.T) {
 			fields: fieldsTestDeleteCredentialCmd{
 				credDelete: credDeleteMock{},
 				reader:     credSettingsMock{},
-				ctxFinder: ctxFinderCustomMock{
-					findMock: func() (rcontext.ContextHolder, error) {
-						return rcontext.ContextHolder{}, errors.New("some error on find Context")
+				envFinder: envFinderCustomMock{
+					find: func() (env.Holder, error) {
+						return env.Holder{}, errors.New("some error on find Context")
```
Comment (brunasilvazup) [inline pkg/cmd/delete_credential_test.go:77]: `some error on find Context ` -> `some error on find Env`

File: pkg/cmd/delete_env_test.go [hunk 5]
```diff
@@ -18,13 +18,19 @@ package cmd
 
 import (
 	"testing"
+
+	"github.com/ZupIT/ritchie-cli/pkg/env"
 )
 
 func TestNewDeleteContextCmd(t *testing.T) {
```
Comment (brunasilvazup) [inline pkg/cmd/delete_env_test.go:25]: `TestNewDeleteContextCmd -> `TestNewDeleteEnvCmd`

File: pkg/cmd/mocks_test.go [hunk 6]
```diff
@@ -207,43 +208,45 @@ func (workspaceForm) Validate(workspace formula.Workspace) error {
 
 type ctxSetterMock struct{}
```
Comment (brunasilvazup) [inline pkg/cmd/mocks_test.go:209]: This function should be `envSetterMock` right?

File: pkg/cmd/mocks_test.go [hunk 7]
```diff
@@ -207,43 +208,45 @@ func (workspaceForm) Validate(workspace formula.Workspace) error {
...
 
-func (ctxSetterMock) Set(ctx string) (rcontext.ContextHolder, error) {
-	return rcontext.ContextHolder{}, nil
+func (ctxSetterMock) Set(ctx string) (env.Holder, error) {
```
Comment (brunasilvazup) [inline pkg/cmd/mocks_test.go:211]: `ctx string` -> `env string`

File: pkg/cmd/mocks_test.go [hunk 8]
```diff
@@ -207,43 +208,45 @@ func (workspaceForm) Validate(workspace formula.Workspace) error {
...
-func (ctxFindRemoverMock) Find() (rcontext.ContextHolder, error) {
-	f := ctxFinderMock{}
-	return f.Find()
+func (e envFindRemoverMock) Find() (env.Holder, error) {
+	return e.holder, e.err
 }
 
-func (ctxFindRemoverMock) Remove(ctx string) (rcontext.ContextHolder, error) {
-	return rcontext.ContextHolder{}, nil
+func (envFindRemoverMock) Remove(ctx string) (env.Holder, error) {
```
Comment (brunasilvazup) [inline pkg/cmd/mocks_test.go:238]: `ctx string` -> `env string`

File: pkg/cmd/mocks_test.go [hunk 9]
```diff
@@ -207,43 +208,45 @@ func (workspaceForm) Validate(workspace formula.Workspace) error {
...
 
-func (ctxFindSetterMock) Find() (rcontext.ContextHolder, error) {
-	f := ctxFinderMock{}
+func (envFindSetterMock) Find() (env.Holder, error) {
+	f := envFinderMock{}
 	return f.Find()
 }
 
-func (ctxFindSetterMock) Set(ctx string) (rcontext.ContextHolder, error) {
+func (envFindSetterMock) Set(ctx string) (env.Holder, error) {
```
Comment (brunasilvazup) [inline pkg/cmd/mocks_test.go:249]: `ctx string` -> `env string`

File: pkg/cmd/root.go [hunk 10]
```diff
@@ -186,6 +196,48 @@ func tutorialRit(tutorialStatus string) {
 	}
 }
 
+func (ro *rootCmd) convertContextsFileToEnvsFile() error {
```
Comment (brunasilvazup) [inline pkg/cmd/root.go:199]: Do you find it interesting to add a "todo" here?

`// TODO: remove in later releases`

File: pkg/cmd/show_env_test.go [hunk 11]
```diff
@@ -21,9 +21,9 @@ import (
 )
 
 func TestNewShowContextCmd(t *testing.T) {
```
Comment (brunasilvazup) [inline pkg/cmd/show_env_test.go:23]: This function should be `TestNewShowEnvCmd` right?

File: pkg/commands/builder.go [hunk 12]
```diff
@@ -118,21 +116,19 @@ func Build() *cobra.Command {
...
-	ctxRemover := rcontext.NewRemover(ritchieHomeDir, ctxFinder)
-	ctxFindSetter := rcontext.NewFindSetter(ritchieHomeDir, ctxFinder, ctxSetter)
-	ctxFindRemover := rcontext.NewFindRemover(ritchieHomeDir, ctxFinder, ctxRemover)
-	credSetter := credential.NewSetter(ritchieHomeDir, ctxFinder)
+	ctxFinder := env.NewFinder(ritchieHomeDir, fileManager)
+	ctxSetter := env.NewSetter(ritchieHomeDir, ctxFinder, fileManager)
+	ctxRemover := env.NewRemover(ritchieHomeDir, ctxFinder, fileManager)
+	ctxFindSetter := env.NewFindSetter(ctxFinder, ctxSetter)
+	ctxFindRemover := env.NewFindRemover(ctxFinder, ctxRemover)
+	credSetter := credential.NewSetter(ritchieHomeDir, ctxFinder, dirManager, fileManager)
```
Comment (brunasilvazup) [inline pkg/commands/builder.go:124]: Will variable names be changed from `ctx...` to `env...`?

File: pkg/commands/builder.go [hunk 13]
```diff
@@ -215,12 +211,14 @@ func Build() *cobra.Command {
...
-		inputList)
+		inputList,
+	)
 
-	deleteCtxCmd := cmd.NewDeleteContextCmd(ctxFindRemover, inputBool, inputList)
-	setCtxCmd := cmd.NewSetContextCmd(ctxFindSetter, inputText, inputList)
-	showCtxCmd := cmd.NewShowContextCmd(ctxFinder)
+	deleteCtxCmd := cmd.NewDeleteEnvCmd(ctxFindRemover, inputBool, inputList)
+	setCtxCmd := cmd.NewSetEnvCmd(ctxFindSetter, inputText, inputList)
+	showCtxCmd := cmd.NewShowEnvCmd(ctxFinder)
```
Comment (brunasilvazup) [inline pkg/commands/builder.go:219]: Same question here

File: pkg/credential/credential.go [hunk 14]
```diff
@@ -78,6 +78,10 @@ type Pather interface {
 	CredentialsPath() string
 }
 
+type Resolver interface {
```
Comment (brunasilvazup) [inline pkg/credential/credential.go:81]: The `ReadCredentialsValueInContext` function above (line 68) will be changed?

File: pkg/credential/delete_test.go [hunk 15]
```diff
@@ -25,7 +25,7 @@ func (fileRemoverErrorMock) Remove(path string) error {
 func TestCredDelete(t *testing.T) {
 	type args struct {
 		homePath string
-		cf       rcontext.Finder
+		cf       env.Finder
```
Comment (brunasilvazup) [inline pkg/credential/delete_test.go:28]: `cf` -> `ef`

File: pkg/credential/resolver_test.go [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+
+	"github.com/ZupIT/ritchie-cli/pkg/prompt"
+	"github.com/ZupIT/ritchie-cli/pkg/stream"
+)
+
+func TestCredentialResolver(t *testing.T) {
+	fileManager := stream.NewFileManager()
+	dirManager := stream.NewDirManager(fileManager)
+	tempDirectory := os.TempDir()
+	contextFinder := env.NewFinder(tempDirectory, fileManager)
```
Comment (brunasilvazup) [inline pkg/credential/resolver_test.go:34]: `contextFinder` -> `envFinder`?

File: pkg/credential/setter_test.go [hunk 17]
```diff
@@ -63,20 +64,20 @@ func (suite *SetterTestSuite) fileInfo(path string) (string, error) {
 	return string(b), err
 }
 
-func (suite *SetterTestSuite) TestSetCredentialToDefalt() {
+func (suite *SetterTestSuite) TestSetCredentialToDefault() {
 	for _, t := range []struct {
 		testName string
-		context  rcontext.ContextHolder
+		context  env.Holder
```
Comment (brunasilvazup) [inline pkg/credential/setter_test.go:70]: `context` -> `env`

File: pkg/credential/setter_test.go [hunk 18]
```diff
@@ -63,20 +64,20 @@ func (suite *SetterTestSuite) fileInfo(path string) (string, error) {
...
 	}{
-		{"Context informed", *suite.contextHolderDefault},
-		{"Context not informed", *suite.contextHolderNil},
+		{"Context informed", *suite.envHolderDefault},
+		{"Context not informed", *suite.envHolderNil},
```
Comment (brunasilvazup) [inline pkg/credential/setter_test.go:73]: `"Context...` -> `"Env...`

File: pkg/credential/setter_test.go [hunk 19]
```diff
@@ -63,20 +64,20 @@ func (suite *SetterTestSuite) fileInfo(path string) (string, error) {
...
 	} {
 		suite.Run(t.testName, func() {
 			contextFinderMock := new(mocks.ContextFinderMock)
```
Comment (brunasilvazup) [inline pkg/credential/setter_test.go:76]: `contextFinderMock` -> `envFinderMock`

File: pkg/formula/runner/docker/runner.go [hunk 20]
```diff
@@ -149,18 +150,18 @@ func (ru RunManager) runDocker(setup formula.Setup, inputType api.TermInputType,
...
-	ctx, err := ru.ctx.Find()
+	envHolder, err := ru.env.Find()
 	if err != nil {
 		return err
 	}
 
 	dockerEnv := fmt.Sprintf(formula.EnvPattern, formula.DockerExecutionEnv, "true")
 	pwdEnv := fmt.Sprintf(formula.EnvPattern, formula.PwdEnv, pwd)
-	ctxEnv := fmt.Sprintf(formula.EnvPattern, formula.CtxEnv, ctx.Current)
+	ctxEnv := fmt.Sprintf(formula.EnvPattern, formula.CtxEnv, envHolder.Current)
```
Comment (brunasilvazup) [inline pkg/formula/runner/docker/runner.go:160]: `ctxEnv` -> `envEnv`

img

Comment (brunasilvazup) [reply pkg/formula/runner/docker/runner.go:160]: And `formula.CtxEnv` will it be changed?

File: pkg/formula/runner/docker/runner_test.go [hunk 21]
```diff
@@ -51,12 +50,12 @@ func TestRun(t *testing.T) {
 	zipFile := filepath.Join("..", "..", "..", "..", "testdata", "ritchie-formulas-test.zip")
 	_ = streams.Unzip(zipFile, repoPath)
 
-	ctxFinder := rcontext.NewFinder(ritHome, fileManager)
+	ctxFinder := env.NewFinder(ritHome, fileManager)
```
Comment (brunasilvazup) [inline pkg/formula/runner/docker/runner_test.go:53]: `ctxFinder` -> `envFinder`

File: pkg/formula/runner/docker/runner_test.go [hunk 22]
```diff
@@ -70,7 +69,7 @@ func TestRun(t *testing.T) {
 		preRun        formula.PreRunner
 		postRun       formula.PostRunner
 		inputResolver formula.InputResolver
-		context       rcontext.Finder
+		context       env.Finder
```
Comment (brunasilvazup) [inline pkg/formula/runner/docker/runner_test.go:72]: `context` -> `env` ?

File: pkg/formula/runner/docker/runner_test.go [hunk 23]
```diff
@@ -263,11 +262,11 @@ func (i inputTextDefaultMock) Text(formula.Input) (string, error) {
 }
 
 type ctxFinderMock struct {
-	ctx rcontext.ContextHolder
+	ctx env.Holder
```
Comment (brunasilvazup) [inline pkg/formula/runner/docker/runner_test.go:265]: `ctxFinderMock` -> `envFinderMock` and `ctx` -> `env`

File: pkg/formula/runner/local/runner_test.go [hunk 24]
```diff
@@ -53,12 +52,12 @@ func TestRun(t *testing.T) {
 	zipFile := filepath.Join("..", "..", "..", "..", "testdata", "ritchie-formulas-test.zip")
 	_ = streams.Unzip(zipFile, repoPath)
 
-	ctxFinder := rcontext.NewFinder(ritHome, fileManager)
+	ctxFinder := env.NewFinder(ritHome, fileManager)
```
Comment (brunasilvazup) [inline pkg/formula/runner/local/runner_test.go:55]: `ctxFinder` -> `envFinder`

File: pkg/formula/runner/local/runner_test.go [hunk 25]
```diff
@@ -73,7 +72,7 @@ func TestRun(t *testing.T) {
 		postRun       formula.PostRunner
 		inputResolver formula.InputResolver
 		fileManager   stream.FileWriteExistAppender
-		context       rcontext.Finder
+		context       env.Finder
```
Comment (brunasilvazup) [inline pkg/formula/runner/local/runner_test.go:75]: `context` -> `env`

File: pkg/formula/runner/local/runner_test.go [hunk 26]
```diff
@@ -251,11 +250,11 @@ func (i inputTextDefaultMock) Text(formula.Input) (string, error) {
 }
 
 type ctxFinderMock struct {
-	ctx rcontext.ContextHolder
+	ctx env.Holder
```
Comment (brunasilvazup) [inline pkg/formula/runner/local/runner_test.go:253]: `ctxFinderMock` -> `envFinderMock`

File: pkg/formula/runner/docker/runner.go (hunk 20 above)
Comment (henriquemoraeszup) [reply pkg/formula/runner/docker/runner.go:160]: maybe simply env?

File: pkg/credential/credential.go (hunk 14 above)
Comment (kaduartur) [reply pkg/credential/credential.go:81]: yes, I will change it.

File: pkg/credential/resolver_test.go [hunk 27]
```diff
@@ -0,0 +1,149 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			credentialResolver := NewResolver(tt.in.credFinder, tt.in.credSetter, tt.in.pass)
+			got, err := credentialResolver.Resolve(tt.in.credentialField)
+
+			if tt.out.err != nil && tt.out.err.Error() != err.Error() {
+				t.Errorf("Resolve(%s) got %v, want %v", tt.name, err, tt.out.err)
+			}
+			if got != tt.out.want {
+				t.Errorf("Resolve(%s) got %v, want %v", tt.name, got, tt.out.want)
+			}
```
Comment (henriquemoraeszup) [inline pkg/credential/resolver_test.go:129]: can we use testify assertions?

File: pkg/credential/resolver_test.go [hunk 28]
```diff
@@ -0,0 +1,149 @@
...
+	return pass.value, pass.err
+}
+
+type setCredMock struct {
+	err error
+}
+
+func (s setCredMock) Set(d Detail) error {
+	return s.err
+}
```
Comment (henriquemoraeszup) [inline pkg/credential/resolver_test.go:149]: can we use testify mocks? From the mock package prefereably

**Review by henriquemoraeszup — DISMISSED** [review]
You can ignore my test comments, it might not be the scope of this pr

File: pkg/credential/resolver_test.go (hunk 27 above)
Comment (kaduartur) [reply pkg/credential/resolver_test.go:129]: makes sense, but I think we can do it in another task :)

**Review by victorschumacherzup — APPROVED** [review]

**Review by henriquemoraeszup — APPROVED** [review]
🚀

**Pull request merged**